File: 01/cpp/insert.cpp
#include <iostream>
#include "testHelper.h"
#include "algorithm.h"

using namespace std;

int main() {
	int n = 10000;
	// int *arr = TestHelper::generateRandomArray(n, 0, n);
	int *arr = TestHelper::generateNearlyOrderedArray(n, 10);
	int *arr2 = TestHelper::copyIntArray(arr, n);

	TestHelper::testSort("insertion sort", Algorithm::insertionSort, arr, n);
	TestHelper::testSort("selection sort", Algorithm::selectionSort, arr2, n);

	delete[] arr;
	delete[] arr2;

	return 0;
}

File: 01/js/insert.js
const TestHelper = require('./testHelper.js');
const Algorithm = require('./algorithm.js');

const n = 10000;
// const arr = TestHelper.generateRandomArray(n, 0, n);
const arr = TestHelper.generateNearlyOrderedArray(n, 10);
const arr2 = TestHelper.copyArray(arr);

TestHelper.testSort('insertion sort', Algorithm.insertionSort, arr);
TestHelper.testSort('selection sort', Algorithm.selectionSort, arr2);
